src/main/scala/scala/meta/serialiser/mappable.scala

```diff
@@ -5,8 +5,9 @@ import scala.annotation.StaticAnnotation
 import scala.collection.immutable.Seq
 import scala.meta._
 
-// type classes that mappable will generate
+// type classes that mappable will generate for annotated classes
 trait ToMap[A] { def apply(a: A): Map[String, Any] }
+trait FromMap[A] { def apply(keyValues: Map[String, Any]): Option[A] }
```

Comment:
we could add `Ops class` to make this easier to use.

likes
```
implicit class ToMapOps[T](c: T) {
  def toMap(implicit m: ToMap[T]): Map[String, Any] = m.toMap(c)
}

def fromMap(m: Map[String, Any])(implicit m: FromMap[T]) = m.fromMap(m)
```

so we can use `someclass.toMap` or `fromMap(m)` directly.